homeassistant/components/motionblinds_ble/sensor.py

```diff
@@ -0,0 +1,221 @@
+"""Sensor entities for the Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from math import ceil
+
+from motionblindsble.const import (
+    MotionBlindType,
+    MotionCalibrationType,
+    MotionConnectionType,
+)
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    ATTR_BATTERY,
+    ATTR_CALIBRATION,
+    ATTR_CONNECTION,
+    ATTR_SIGNAL_STRENGTH,
+    CONF_MAC_CODE,
+    DOMAIN,
+    ICON_CALIBRATION,
+    ICON_CONNECTION,
+)
+from .entity import MotionblindsBLEEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+@dataclass(frozen=True, kw_only=True)
+class MotionblindsBLESensorEntityDescription(SensorEntityDescription):
+    """Entity description of a sensor entity with native_value attribute."""
+
+    native_value: str | None = None
+
+
+SENSOR_TYPES: dict[str, MotionblindsBLESensorEntityDescription] = {
+    ATTR_BATTERY: MotionblindsBLESensorEntityDescription(
+        key=ATTR_BATTERY,
+        translation_key=ATTR_BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
```

Comment:
The device class will handle the name and icons for this entity. Please remove the icon stuff and the translation key.

Reply:
For the battery I also used icons to show whether or not the motor is charging, which I don't think is included in the device class behavior

Reply:
It will be shown in the UI and if this would be a feature I would almost suggest to make this a default way of handling all battery icons

Reply:
I don't think many devices will have access to this information

From the motor feedback I can deduce if the motor is charging or if the motor is wired, which will be reflected in the battery icon

Reply:
But if a motor is wired, what's the benefit of showing a battery percentage?

Reply:
Are certain motors permanently wired? Or is it a device you can carry around or plug in

Reply:
Some motors are permanently 230V wired. They do not contain a battery

Reply:
Is there a way we can listen to the callback in `async_setup_entry` and if we receive that it's not wired, we create the entity and unsubscribe?

Reply:
The status message is only sent once the motor is connected to, therefore I think doing something like that would be hard. The only option would be to include it later in the config flow, but that may be a bit much

Reply:
Hmmmm, interesting one

Reply:
Indeed, but I don't really think there is a better option than just showing 100% for the battery and changing it's icon to a plug to indicate it is wired